kube/objects.go

```diff
@@ -268,6 +268,39 @@ func objectsWithoutNils(objects *Objects) *Objects {
 	return objects
 }
 
+func NewClientConfigFromKubeConfig(opts *options) (*rest.Config, error) {
+	config := &rest.Config{}
+
+	err := opts.validate()
+	if err != nil {
+		return nil, err
+	}
+
+	if opts.yaml != nil {
+		config, err = clientcmd.RESTConfigFromKubeConfig(opts.yaml)
+	} else {
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		if len(opts.paths) != 0 {
+			loadingRules.Precedence = opts.paths
+		}
+		configOverrides := &clientcmd.ConfigOverrides{}
+		if opts.kubeContext != "" {
+			configOverrides.CurrentContext = opts.kubeContext
+		}
+		config, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides).ClientConfig()
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	config.Timeout = opts.timeout
```

Comment:
I think the timeout config applies even to in-cluster config, so this should be moved down to NewClient. Same applies for transportWrapper